services/cd-service/pkg/service/batch.go

```diff
@@ -92,12 +93,37 @@ func ValidateApplication(
 	return nil
 }
 
+func (d *BatchServer) checkUserPermissions(user *auth.User, act *api.CreateEnvironmentLockRequest, application, action string) error {
```

Comment:
```suggestion
func (d *BatchServer) checkUserPermissions(user *auth.User, env string, application, action string) error {
```
You're only using the environment.